mantle/platform/conf/conf.go

```diff
@@ -332,6 +332,76 @@ func (c *Conf) MergeV34exp(newConfig v34exptypes.Config) {
 	c.ignitionV34exp = &mergeConfig
 }
 
+// Merge all configs into a V3.4 config
+func MergeUserData(userDataObjs []*UserData) (*UserData, error) {
```

Comment:
With `RenderToV34exp()`, I think we could have this instead take an array of `Conf`s and just call the existing `MergeV34exp()` in a loop. We leave it to the caller to use `RenderToV34exp` for all the configs it intends to merge together and then we can get rid of `upgradeAndMerge`.